plugins/pagerduty/src/components/PagerDutyCard/index.tsx

```diff
@@ -67,33 +76,54 @@ export const PagerDutyCard = () => {
     loading,
     error,
   } = useAsync(async () => {
-    const services = await api.getServiceByIntegrationKey(
-      integrationKey as string,
-    );
+    let service: Service;
+
+    if (integrationKey) {
+      const services = await api.getServiceByIntegrationKey(
+        integrationKey as string,
+      );
+      service = services[0];
+      if (!service) throw new NotFoundError();
+    } else {
+      service = await api.getServiceByServiceId(serviceId);
+    }
 
     return {
-      id: services[0].id,
-      name: services[0].name,
-      url: services[0].html_url,
-      policyId: services[0].escalation_policy.id,
-      policyLink: services[0].escalation_policy.html_url,
+      id: service.id,
+      name: service.name,
+      url: service.html_url,
+      policyId: service.escalation_policy.id,
+      policyLink: service.escalation_policy.html_url,
     };
   });
 
-  if (error instanceof UnauthorizedError) {
-    return <MissingTokenError />;
-  }
-
   if (error) {
-    return (
-      <Alert severity="error">
-        Error encountered while fetching information. {error.message}
-      </Alert>
-    );
+    let errorNode: ReactNode;
+
+    switch (error.constructor) {
+      case UnauthorizedError:
+        errorNode = <MissingTokenError />;
+        break;
+      case NotFoundError:
+        errorNode = <ServiceNotFoundError />;
+        break;
+      default:
+        errorNode = (
+          <Alert severity="error">
+            Error encountered while fetching information. {error.message}
+          </Alert>
+        );
+    }
+
+    return <BasicCard>{errorNode}</BasicCard>;
```

Comment:
wrapping the empty state errors in this `BasicCard` _does_ unfortunately have an undesirable "dark background over light background over dark background" effect when in dark mode. Happy to remove this for errors, but the consistent Info Card seemed desirable